salt/modules/cmdmod.py

```diff
@@ -651,21 +651,12 @@ def _get_stripped(cmd):
         except (OSError, IOError) as exc:
             msg = (
                 'Unable to run command \'{0}\' with the context \'{1}\', '
-                'reason: '.format(
+                'reason: {2}'.format(
                     cmd if output_loglevel is not None else 'REDACTED',
-                    new_kwargs
+                    new_kwargs,
+                    exc.strerror
```

Comment:
I would personally replace this with `exc`. This will force `.format()` to stringify it, which will in addition to the error message give us the error code. In addition, it will insulate this `except` block against potential future issues, since not all exception classes have a `strerror` attribute.

Reply:
@terminalmage - okay, I will do that tomorrow morning. I had the same thing in mind, but exc will print: 'OSError: [Errno 12] Cannot allocate memory' as the 'reason', and I saw in official python docs that both OSError and IOError do have the 'strerror' attribute (but it can be None obviously for certain exceptions). Basically both OSError and IOError are constructed from an int and a string, the int being the errno, and the string the explanations for it.

Reply:
@rares-pop I think it will just show `[Errno 12] Cannot allocate memory`.

But my main concern here is that someone later adds some other exception class to that `except` and then causes an `AttributeError`.

I'm not saying you did anything terribly wrong, just trying to make it a bit more future-proof.